modules/plugin/geopkg/src/test/java/org/geotools/geopkg/GeoPkgFilterToSQLTest.java

```diff
@@ -0,0 +1,53 @@
+package org.geotools.geopkg;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.StringWriter;
+import org.geotools.data.jdbc.SQLFilterTestSupport;
+import org.geotools.factory.CommonFactoryFinder;
+import org.geotools.feature.SchemaException;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengis.feature.IllegalAttributeException;
+import org.opengis.filter.FilterFactory2;
+import org.opengis.filter.PropertyIsLike;
+
+public class GeoPkgFilterToSQLTest extends SQLFilterTestSupport {
+
+    private static FilterFactory2 ff;
+
+    private GeoPkgDialect dialect;
+
+    GeoPkgFilterToSQL filterToSql;
+
+    StringWriter writer;
+
+    @Override
+    @Before
+    public void setUp() throws IllegalAttributeException, SchemaException {
+        ff = CommonFactoryFinder.getFilterFactory2();
+        dialect = new GeoPkgDialect(null);
+        filterToSql = new GeoPkgFilterToSQL(dialect);
+        writer = new StringWriter();
+        filterToSql.setWriter(writer);
+
+        prepareFeatures();
+    }
+
+    @Test
+    public void testFunctionLike() throws Exception {
+        filterToSql.setFeatureType(testSchema);
+        PropertyIsLike like =
+                ff.like(
+                        ff.function("strToLowerCase", ff.property("testString")),
+                        "a_literal",
+                        "%",
+                        "-",
+                        "\\",
+                        true);
+
+        filterToSql.encode(like);
```

Comment:
So before the change in FilterToSQL this line would have thrown a "Filter type not supported" exception right?

Reply:
Yes, it was throwing that exception.